clang/lib/Driver/Driver.cpp

```diff
@@ -5203,761 +5225,812 @@ class OffloadingActionBuilder final {
       }
     }
 
-    // Return whether to use native bfloat16 library.
-    bool selectBfloatLibs(const ToolChain *TC, bool &useNative) {
-      const char *TargetOpt = nullptr;
-      const char *DeviceOpt = nullptr;
-      bool needLibs = false;
-      for (auto *A : Args) {
-        llvm::Triple *TargetBE = nullptr;
-
-        auto GetTripleIt = [&, this](llvm::StringRef Triple) {
-          llvm::Triple TargetTriple{Triple};
-          auto TripleIt = llvm::find_if(SYCLTripleList, [&](auto &SYCLTriple) {
-            return SYCLTriple == TargetTriple;
-          });
-          return TripleIt != SYCLTripleList.end() ? &*TripleIt : nullptr;
-        };
+    // Performs device specific linking steps for the SYCL based toolchain.
+    // This function is used for both the early AOT flow and the typical
+    // offload device link flow.
+    // The ListIndex input provides an index against the SYCLTargetInfoList
+    // used to determine associated toolchain information for the values being
+    // worked against to add the device link steps.  This is also used to change
+    // how the toolchain is formed, either by dependency additions or by adding
+    // actions explicitly.
+    void appendSYCLDeviceLink(ActionList &LI, const ToolChain *TC,
+                              OffloadAction::DeviceDependences &DA,
+                              ActionList &AL, const char *BoundArch,
+                              bool addOffloadAction = false) {
+      auto addDeps = [&](Action *A, const ToolChain *TC,
+                         const char *BoundArch) {
+        if (addOffloadAction) {
+          OffloadAction::DeviceDependences Deps;
+          Deps.add(*A, *TC, BoundArch, Action::OFK_SYCL);
+          AL.push_back(C.MakeAction<OffloadAction>(Deps, A->getType()));
+        } else {
+          DA.add(*A, *TC, BoundArch, Action::OFK_SYCL);
+        }
+      };
 
-        if (A->getOption().matches(options::OPT_fsycl_targets_EQ)) {
-          // spir64 target is actually JIT compilation, so we defer selection of
-          // bfloat16 libraries to runtime. For AOT we need libraries, but skip
-          // for Nvidia.
-          needLibs = TC->getTriple().getSubArch() != llvm::Triple::NoSubArch &&
-                     !TC->getTriple().isNVPTX();
-          TargetBE = GetTripleIt(A->getValue(0));
-          if (TargetBE)
-            TargetOpt = A->getValue(0);
-          else
+      // List of device specific libraries to be fed into llvm-link.
+      ActionList DeviceLibs;
+
+      // List of device specific library 'objects' (FPGA AOCO libraries) that
+      // are fed directly to the FPGA offline compiler.
+      ActionList DeviceLibObjects;
+
+      // List of device objects that go through the device link step.
+      ActionList LinkObjects;
+      auto TT = TC->getTriple();
```

Comment:
Nit. A more descriptive name can be helpful. 
[LLVM Coding Std](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly)

Reply:
Given the broad usage, I agree, will update